src/c_api/c_predict_api.cc

```diff
@@ -210,19 +249,31 @@ int _CreatePartialOut(const char* symbol_json_str,
 
   std::vector<NDArray> arg_arrays, aux_arrays;
   for (size_t i = 0; i < arg_shapes.size(); ++i) {
-    NDArray nd = NDArray(arg_shapes[i], ctx);
+    NDArray nd;
+    if (result_arg_types[i] != -1) {
+      nd = NDArray(arg_shapes[i], ctx, false, result_arg_types[i]);
+    } else {
+      nd = NDArray(arg_shapes[i], ctx);
+    }
     if (arg_params.count(arg_names[i]) != 0) {
       CopyFromTo(arg_params[arg_names[i]], &nd);
     }
     arg_arrays.push_back(nd);
   }
+
   for (size_t i = 0; i < aux_shapes.size(); ++i) {
-    NDArray nd = NDArray(aux_shapes[i], ctx);
+    NDArray nd;
+    if (result_aux_types[i] != -1) {
```

Comment:
Since we are doing such check on types throughout, can we otherwise think of a setting a default DType for all params if users are not providing arg types params? So we could get rid of all these checks and always types will be available. Thoughts?

Reply:
This was added as I was foreseeing the AMP change. and didnt even have the 
```
    CHECK(infer_type_complete)
        << "The type information is not enough, please provide input arg_types "
           "with provided_arg_dtype_names and provided_arg_dtypes";
```
Will add a test for AMP for C Predict API and then get back here.

Reply:
Thanks.

Reply:
i have added the tests now.